src/controllers/media.ts

```diff
@@ -338,3 +338,147 @@ export const getByImdbID = async(ctx: ParameterizedContext<any, Router.IRouterPa
+
+  if (osdbHash) {
+    query.push({ osdbHash });
+    failedQuery.push({ osdbHash });
+    if (!imdbID && !title) {
+      isOnlyOpenSubsSearch = true;
+    }
+  }
+
+  if (imdbID) {
+    query.push({ imdbID });
+    failedQuery.push({ imdbId: imdbID });
+  }
+
+  if (title) {
+    const titleQuery: any = { searchMatches: { $in: [title] } };
+    const titleFailedQuery: any = { title };
+
+    if (year) {
+      titleQuery.year = year;
+      titleFailedQuery.year = year;
+    }
+    if (episode) {
+      titleQuery.episode = episode;
+      titleFailedQuery.episode = episode;
+    }
+    if (season) {
+      titleQuery.season = season;
+      titleFailedQuery.season = season;
+    }
+    query.push(titleQuery);
+    failedQuery.push(titleFailedQuery);
+  }
+
+  // find an existing metadata record, or previous failure record
+  const [existingResult, existingFailedResult] = await Promise.all([
+    MediaMetadata.findOne({ $or: query }, null, { lean: true }).exec(),
+    FailedLookups.findOne({ $or: failedQuery }, null, { lean: true }).exec(),
+  ]);
+
+  // we have an existing metadata record, so return it
+  if (existingResult) {
+    return ctx.body = existingResult;
+  }
+
+  // we have an existing failure record, so increment it, and throw not found error
+  if (existingFailedResult) {
+    await FailedLookups.updateOne({ _id: existingFailedResult._id }, { $inc: { count: 1 } }).exec();
+    throw new MediaNotFoundError();
+  }
+
+  // the database does not have a record of this file, so begin search for metadata on external apis.
+
+  // Start OpenSubtitles lookups
+  let openSubtitlesMetadata;
+
+  if (osdbHash && filebytesize) {
+    const osQuery = { moviehash: osdbHash, moviebytesize: filebytesize };
+    const validation = {
+      year: year ? year : null,
+      season: season ? season : null,
+      episode: episode ? episode : null,
+    };
+
+    openSubtitlesMetadata = await externalAPIHelper.getFromOpenSubtitles(osQuery, validation);
+
+    if (isOnlyOpenSubsSearch && !openSubtitlesMetadata) {
+      await FailedLookups.updateOne({ osdbHash }, { $inc: { count: 1 } }, { upsert: true, setDefaultsOnInsert: true }).exec();	
+      throw new MediaNotFoundError();
+    }
+  }
+
+  // End OpenSubtitles lookups
+
+  /* if the client has passed an imdbId, we'll use that as the primary source, otherwise, the one
+  returned by OpenSubtitles, if one was found */
+
+  // Start omdb lookups
+  const omdbSearchRequest = {} as SearchRequest;
+  const imdbIdToSearch = imdbID ? imdbID
+    : openSubtitlesMetadata?.imdbID ? openSubtitlesMetadata.imdbID : null;
+
+  // if the client did not pass an imdbID, but we found one from Open Subtitles, so see if we have an existing record for the now known media.
+  if (!imdbID && openSubtitlesMetadata?.imdbID) {
+    {
+      const existingResult = await MediaMetadata.findOne({ imdbID }, null, { lean: true }).exec();
+      if (existingResult) {
+        return ctx.body = existingResult;
+      }
+    }
```

Comment:
```suggestion
    const existingResult = await MediaMetadata.findOne({ imdbID }, null, { lean: true }).exec();
    if (existingResult) {
      return ctx.body = existingResult;
    }
```

Reply:
This was intended - I wanted to use the same variable name so this creates a new scope. I can change the name variable and remove it, but maybe you just thought i didnt mean to do that

Reply:
All good, yeah I didn't realise